python/fbprophet/plot.py

```diff
@@ -672,3 +673,303 @@ def plot_plotly(m, fcst, uncertainty=True, plot_cap=True, trend=False, changepoi
+        for trace in components[name]['traces']:
+            fig.append_trace(trace, i + 1, 1)
+    return fig
+
+
+def plot_forecast_component_plotly(m, fcst, name, uncertainty=True, plot_cap=False, figsize=(900, 300)):
+    """Plot an particular component of the forecast using Plotly.
+    See plot_plotly() for Plotly setup instructions
+
+    Parameters
+    ----------
+    m: Prophet model.
+    fcst: pd.DataFrame output of m.predict.
+    name: Name of the component to plot.
+    uncertainty: Optional boolean to plot uncertainty intervals.
+    plot_cap: Optional boolean indicating if the capacity should be shown
+        in the figure, if available.
+    figsize: The plot's size (in px).
+
+    Returns
+    -------
+    A Plotly Figure.
+    """
+    props = get_forecast_component_plotly_props(m, fcst, name, uncertainty, plot_cap)
+    layout = go.Layout(
+        width=figsize[0],
+        height=figsize[1],
+        showlegend=False,
+        xaxis=props['xaxis'],
+        yaxis=props['yaxis']
+    )
+    fig = go.Figure(data=props['traces'], layout=layout)
+    return fig
+
+
+def plot_seasonality_plotly(m, name, uncertainty=True, figsize=(900, 300)):
+    """Plot a custom seasonal component using Plotly.
+    See plot_plotly() for Plotly setup instructions
+
+    Parameters
+    ----------
+    m: Prophet model.
+    name: Seasonality name, like 'daily', 'weekly'.
+    uncertainty: Optional boolean to plot uncertainty intervals.
+    figsize: Set the plot's size (in px).
+
+    Returns
+    -------
+    A Plotly Figure.
+    """
+    props = get_seasonality_plotly_props(m, name, uncertainty)
+    layout = go.Layout(
+        width=figsize[0],
+        height=figsize[1],
+        showlegend=False,
+        xaxis=props['xaxis'],
+        yaxis=props['yaxis']
+    )
+    fig = go.Figure(data=props['traces'], layout=layout)
+    return fig
+
+
+def get_forecast_component_plotly_props(m, fcst, name, uncertainty=True, plot_cap=False):
+    """Prepares a dictionary for plotting the selected forecast component with Plotly
+
+    Parameters
+    ----------
+    m: Prophet model.
+    fcst: pd.DataFrame output of m.predict.
+    name: Name of the component to plot.
+    uncertainty: Optional boolean to plot uncertainty intervals.
+    plot_cap: Optional boolean indicating if the capacity should be shown
+        in the figure, if available.
+
+    Returns
+    -------
+    A dictionary with Plotly traces, xaxis and yaxis
+    """
+    prediction_color = '#0072B2'
+    error_color = 'rgba(0, 114, 178, 0.2)'  # '#0072B2' with 0.2 opacity
+    cap_color = 'black'
+    zeroline_color = '#AAA'
+    line_width = 2
+
+    range_margin = (fcst['ds'].max() - fcst['ds'].min()) * 0.05
+    range_x = [fcst['ds'].min() - range_margin, fcst['ds'].max() + range_margin]
+
+    text = None
+    mode = 'lines'
+    if name == 'holidays':
+        fcst = fcst[fcst[name] != 0].copy()
+        text = fcst.merge(m.holidays, how='left', on='ds')['holiday']
```

Comment:
This runs into issues when there are multiple holidays that share a date, and also has slightly odd behavior with the lower and upper windows on holidays.

You can produce these issues of you generate this plot on the holiday example at the top of https://github.com/facebook/prophet/blob/master/notebooks/seasonality%2C_holiday_effects%2C_and_regressors.ipynb . In this example, each holiday as a +1 holiday window, which means the day after the holiday is also treated as a holiday. What you can see in the first year of the holidays plot is that in these cases "playoff" has the correct text label, but the next day has no text label. This is because it's date is not actually in m.holidays. But the bigger issue is with holidays that share a date. In this case "superbowl" is a subset of "playoff". On these overlapping days, the value being plotted is the sum of the effects. It looks like the text labels are getting mixed up, so that after the first occurrence of "superbowl" some of the "playoff" points are being labeled as "superbowl", and vice versa. 

I think the text labels are _really_ cool and quite useful. So if it is possible to fix them that'd be great, but if you prefer to get this in sooner you could remove them for now and add it back in once it's fixed.

Reply:
Well spotted. I've added some new logic that should fix the wrong placements and also include the windows.